python/lsst/fgcmcal/focalPlaneProjector.py

```diff
@@ -0,0 +1,147 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""A class to project the focal plane in arbitrary rotations for fgcm.
+
+This file contains a class used by fgcm ...
+"""
+from functools import lru_cache
+import numpy as np
+
+import lsst.afw.image as afwImage
+import lsst.afw.cameraGeom as afwCameraGeom
+import lsst.geom as geom
+from lsst.obs.base import createInitialSkyWcs
+
+__all__ = ['FocalPlaneProjector']
+
+
+class FocalPlaneProjector(object):
+    """
+    Class to project the focal plane onto the sky.
+
+    Parameters
+    ----------
+    camera : `lsst.afw.cameraGeom.Camera`
+        Camera from the butler.
+    defaultOrientation : `int`
+        Default camera orientation in degrees.
+    """
+    def __init__(self, camera, defaultOrientation):
+        self.camera = camera
+
+        # Put the reference boresight at the equator to avoid cos(dec) problems.
+        self.boresight = geom.SpherePoint(180.0*geom.degrees, 0.0*geom.degrees)
+        self.flipX = False
+        self.defaultOrientation = int(defaultOrientation)
+
+    def _makeWcsDict(self, orientation):
+        """
+        Make a dictionary of WCSs at the reference boresight position.
+
+        Parameters
+        ----------
+        orientation : `int`
+            Orientation, to the nearest degree.
+
+        Returns
+        -------
+        wcsDict : `dict`
+            Dictionary of WCS, with the detector id as the key.
+        """
+        _orientation = orientation*geom.degrees
+
+        visitInfo = afwImage.VisitInfo(boresightRaDec=self.boresight,
+                                       boresightRotAngle=_orientation,
+                                       rotType=afwImage.RotType.SKY)
+
+        wcsDict = {}
+
+        for i, detector in enumerate(self.camera):
+            detectorId = detector.getId()
+            wcsDict[detectorId] = createInitialSkyWcs(visitInfo, detector, self.flipX)
+
+        return wcsDict
+
+    @lru_cache(maxsize=360)
+    def __call__(self, orientation, nstep=50):
+        """
+        Make a focal plane projection mapping for use with fgcm.
+
+        Parameters
+        ----------
+        orientation : `int`
+            Camera orientation in degrees.  Integerized for better caching.
```

Comment:
Can orientation be defined?  Is this the angle between camera +y and North, for instance?

Reply:
I'll put this in:
https://github.com/lsst/afw/blob/7e580e471ce3e8189aece5981cba78b6b9ec6862/include/lsst/afw/image/VisitInfo.h#L170-L171